x-pack/plugins/security/server/routes/index.ts

```diff
@@ -74,6 +75,7 @@ export function defineRoutes(params: RouteDefinitionParams) {
     defineDeprecationsRoutes(params); // deprecated kibana user roles are not applicable, these HTTP APIs are not needed
     defineIndicesRoutes(params); // the ES privileges form used to help define roles (only consumer) is disabled, so there is no need for these HTTP APIs
     defineRoleMappingRoutes(params); // role mappings are managed internally, based on configurations in control plane, these HTTP APIs are not needed
+    defineSecurityFeatureRoutes(params);
```

Comment:
I think it is still ok not to register this endpoint in serverless. But as it is now no longer a role mappings endpoint (never really was), would it benefit us to register it for serverless? Would it simplify the code we had to add to work around needing a "conditional hook" if we could just call the endpoint regardless of build flavor and rely on the response?

cc: @legrego @SiddharthMantri

Reply:
I would keep it out of serverless TBH.
Making endpoint available in serverless changes the scope of this PR a lot. 
Though I'm not sure how much we would leverage with this change, it seems it might require a carefull testing for integration/usage of it and might be a not that trivial in the end. Seems like the benefit of it might be negligible.

Reply:
> I would keep it out of serverless TBH, at least in scope of that PR since it doesn't change access to the endpoint.
>...
> If we consider such change, it would be better to do that in a separate issue/PR.

Sounds good to me. Yeah, I don't think we need to hold up this PR at all. I just wanted to have the discussion.